test/jdk/java/awt/Clipboard/LostOwnershipChainTest/SystemClipboard2ProcTest.java

```diff
@@ -0,0 +1,246 @@
+ *          can set contents of the clipboard and its owner. The clipboard is
+ *          the system clipboard and the owners of the clipboard are in
+ *          2 different processes.
+ * @key headful
+ * @run main SystemClipboard2ProcTest
+*/
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.ClipboardOwner;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class SystemClipboard2ProcTest {
+
+    public static void main(String[] args) throws Exception {
+        SystemClipboardOwner.run();
+
+        if (SystemClipboardOwner.failed) {
+            throw new RuntimeException("test failed: can not get actual " +
+                    "contents of the clipboard or set owner of the clipboard");
+        } else {
+            System.err.println("test passed");
+        }
+    }
+}
+
+class SystemClipboardOwner implements ClipboardOwner {
+    static volatile boolean failed;
+
+    private static final Object LOCK = new Object();
+
+    private static final int CHAIN_LENGTH = 5;
+    private final static Clipboard clipboard =
+        Toolkit.getDefaultToolkit().getSystemClipboard();
+
+    private int m, id;
+
+    public SystemClipboardOwner(int m) { this.m = m; id = m; }
+
+    public void lostOwnership(Clipboard cb, Transferable contents) {
+        System.err.println(id + " lost clipboard ownership");
+
+        Transferable t = getClipboardContents(cb, null);
+        // for test passing if t.getTransferData() will throw an exception
+        String msg = "" + (m + 1);
+        try {
+            msg = (String)t.getTransferData(DataFlavor.stringFlavor);
+        } catch (Exception e) {
+            System.err.println(id + " can't getTransferData: " + e);
+        }
+        System.err.println(id + " Clipboard.getContents(): " + msg);
+        if (!msg.equals("" + (m + 1))) {
+            failed = true;
+            System.err.println("Clipboard.getContents() returned incorrect contents!");
+        }
+
+        m += 2;
+        if (m <= CHAIN_LENGTH) {
+            System.err.println(id + " Clipboard.setContents(): " + m);
+            setClipboardContents(cb, new StringSelection(m + ""), this);
+        }
+        if (m >= CHAIN_LENGTH) {
+            synchronized (LOCK) {
+                LOCK.notifyAll();
+            }
+        }
+    }
+
+    public static void run() throws Exception {
+        SystemClipboardOwner cbo1 = new SystemClipboardOwner(0);
+        System.err.println(cbo1.m + " Clipboard.setContents(): " + cbo1.m);
+        setClipboardContents(clipboard, new StringSelection(cbo1.m + ""),
+                             cbo1);
+
+
+        String javaPath = System.getProperty("java.home", "");
+
+        Process process = new ProcessBuilder(
```

Comment:
Don't we already have some libraries, like ProcessTools and OutputAnalyzer, to run tests in a separate process?
Right now, this test, for example, skips all VM parameters.

Reply:
This is PR opening tests which had this code written a long time ago. Before those libraries existed.
I guess it was a matter of how much effort should be put in to modernise the test, just to get it into open.
Alexander just minimally upgraded it to use ProcessBuilder.

Reply:
@mrserb Thanks for the tip, it allows to remove a lot of duplicated code.

@prrace Changing it to use ProcessTools is quite easy, so updated.

Reply:
it does seem to cut a lot of repeated boiler plate